### PR TITLE
Update to std::path

### PR DESCRIPTION
**Pull request by alexcrichton**

### CODE REVIEW

**Comment by alexcrichton** [comment]
r? @blaenk 

Curious what you think of the fallout in the tests. The most interesting gotcha for me was that `".".join("a")` is `"./a"` instead of `"a"` so I had to add some special logic here and there.

**Comment by blaenk** [comment]
Awww man, sorry about that. I was actually working on this a few weeks ago but got stalled due to missing APIs. @aturon was nice enough to expose them (e.g. `is_separator`, `Prefix::is_verbatim`), but I then forgot about this. I'll take a look right now.

File: src/lib.rs [hunk 1]
```diff
@@ -121,42 +123,57 @@ pub fn glob(pattern: &str) -> Result<Paths, PatternError> {
...
+pub fn glob_with(pattern: &str, options: &MatchOptions)
+                 -> Result<Paths, PatternError> {
     // make sure that the pattern is valid first, else early return with error
     let _compiled = try!(Pattern::new(pattern));
 
     #[cfg(windows)]
-    fn check_windows_verbatim(p: &Path) -> bool { path::windows::is_verbatim(p) }
+    fn check_windows_verbatim(p: &Path) -> bool {
+        use std::path::Prefix;
+        match p.components().next() {
```
Comment (blaenk) [inline src/lib.rs]: Should be able to use [`Prefix::is_verbatim`](http://doc.rust-lang.org/std/path/enum.Prefix.html#method.is_verbatim)

File: src/lib.rs [hunk 2]
```diff
@@ -121,42 +123,57 @@ pub fn glob(pattern: &str) -> Result<Paths, PatternError> {
...
+        }
+    }
     #[cfg(not(windows))]
     fn check_windows_verbatim(_: &Path) -> bool { false }
 
     #[cfg(windows)]
-    fn to_scope(p: Path) -> Path {
-        use std::env::current_dir;
-
-        if path::windows::is_vol_relative(&p) {
```
Comment (blaenk) [inline src/lib.rs]: Looks like I forgot to make sure we had access to `is_vol_relative` in the new path API :/

**Comment by blaenk** [comment]
This is excellent!

I'll try to remember to swap out the `TempDir` for the new one we get, when we get it. I guess I'll create an issue for it.

**Pull request merged**